openstudiocore/src/model/EnergyManagementSystemCurveOrTableIndexVariable.hpp

```diff
@@ -0,0 +1,96 @@
+/**********************************************************************
+ *  Copyright (c) 2008-2016, Alliance for Sustainable Energy.
+ *  All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#ifndef MODEL_ENERGYMANAGEMENTSYSTEMCURVEORTABLEINDEXVARIABLE_HPP
+#define MODEL_ENERGYMANAGEMENTSYSTEMCURVEORTABLEINDEXVARIABLE_HPP
+
+#include <model/ModelAPI.hpp>
+#include "ModelObject.hpp"
+#include "Curve.hpp"
+
+namespace openstudio {
+namespace model {
+
+// TODO: Check the following class names against object getters and setters.
+class AllCurves;
+
+namespace detail {
+
+  class EnergyManagementSystemCurveOrTableIndexVariable_Impl;
+
+} // detail
+
+/** EnergyManagementSystemCurveOrTableIndexVariable is a ModelObject that wraps the OpenStudio IDD object 'OS:EnergyManagementSystem:CurveOrTableIndexVariable'. */
+class MODEL_API EnergyManagementSystemCurveOrTableIndexVariable : public ModelObject {
+ public:
+  /** @name Constructors and Destructors */
+  //@{
+
+  explicit EnergyManagementSystemCurveOrTableIndexVariable(const Model& model, Curve& curve); //Does Curve cover all object-list AllCurves?
+
+  explicit EnergyManagementSystemCurveOrTableIndexVariable(const Model& model);
+
+  virtual ~EnergyManagementSystemCurveOrTableIndexVariable() {}
+
+  //@}
+
+  static IddObjectType iddObjectType();
+
+  /** @name Getters */
+  //@{
+
+  ModelObject curveorTableObject() const;
```

Comment:
Minor thing, I'd prefer curveOrTableObject.  I'll make a few more of these suggestions, I'd poll @kbenne @mbadams5 @asparke2 to see if there is consensus on them